### ISSUE TITLE
Skip commercial cause freezer 

### ISSUE THREAD

**Issue by gsicilia82** [rbuehlma/pvr.zattoo#154]
[kodi.log](https://github.com/rbuehlma/pvr.zattoo/files/9787633/kodi.log)
Hello, I really like to use Kodi instead of official Zattoo app on Fire TV Stick 4k. I am watching Zattoo Switzer and often I use replay function. Unfortunately, really often this causes freezer / jitter / async audio. I think it is caused by fast skip from commercials at the very beginning. 
In the past, I encountered similar issue when I started tvheadend records over Kodi API and jumped directly 3 minutes forward.
Therefore, I can imagine that the issue could be solved if skipping the commercial would start ~5s later. This waiting time fixed my tvheadend record issue. Skipping during Zattoo replays rarely cause this kind of issue. 

I forked your GIT project and changed the EDL timings to start at 5000 instead of 0 ms. I was also able to compile it, but I don't know where to find the installable zip file. And I don't know how to configure compile process for Android/ARM. Maybe you can give me a hint? 

Other idea: 
Is it possible to make the function "skip commercial" optional over addon settings? 

Thank you 

Zattoo Ultimate Switzerland 
Hardware: Fire TV 4K
KODI 19.4
Zattoo PVR Client 19.7.15

Attached file: I started replay at 17:29:01. I could not find any error entry. But also in this case, playback was horrible

**Comment by rbuehlma** [comment]
I suspect this is some issue in Kodi or inputstream.adaptive. To have the root cause solved, probably a good idea is to open against Kodi.

After building the addon, the log should point to the newly created zip file but I suspect that this file will not run on the FireTV if it has not been built for that infrastructure and I have never cross compiled this addon I therefore cannot help you there.

Adding an option to skip the start of a replay show is rather simple. I will do this in the next days but still it would be best to also solve the root cause.

**Comment by gsicilia82** [comment]
I would really appreciate the new option 🙏

About root cause: before zattoo I was using waipu.tv. As I know, waipu uses also widevine. In this case it worked without issues.

I will raise an issue in kodi repo. Maybe, they can forward to responsible projects.

Only for curiosity, do you still use zattoo? If without issues, which platform do you use? Thanks

**Comment by rbuehlma** [comment]
I have just release 19.7.16 with this new setting. Hopefully it solves your issue.

Yes, I'm still using this addon on three different system all on a PC running Ubuntu and am quite happy with it ;)

**Comment by gsicilia82** [comment]
Only as information: Today I tested few different options. Also tested with your latest version. I was able to find the causing option: Audio output with Dolby Digital Plus causes these freezer / jitter. If I chose AAC as audio output, everything works stable. Seeking in both directions works without issues.

This afternoon, I will raise an issue in Inputstream.Adaptive. I could reproduce on my Smartphone, Pixel 6. It seems to be an Android issue. On my Linux server ( Ubuntu x86) it works without a problem.
